data/json/harvest.json

```diff
@@ -2,16 +2,11 @@
   {
     "id": "null",
     "type": "harvest",
-    "entries": []
+    "entries": [  ]
   },
   {
     "id": "fish_small",
     "type": "harvest",
-    "entries": [
-      { "drop": "fish", "scale_num": [ 0.2, 0.4 ], "max": 2 },
-      { "drop": "offal", "base_num": [ 0, 1 ], "scale_num": [ 0.2, 0.4 ], "max": 1 },
-      { "drop": "stomach" },
-      { "drop": "bone", "base_num": [ 1, 2 ], "scale_num": [ 0.4, 0.5 ], "max": 4 }
-    ]
+    "entries": [ { "drop": "fish", "scale_num": [ 0.200000, 0.400000 ], "max": 2 }, { "drop": "offal", "base_num": [ 0, 1 ], "scale_num": [ 0.200000, 0.400000 ], "max": 1 }, { "drop": "stomach" }, { "drop": "bone", "base_num": [ 1, 2 ], "scale_num": [ 0.400000, 0.500000 ], "max": 4 } ]
```

Comment:
The perl script allows configuring whether to break or not. Replicating this would be the most configurable option.
If loading options is too much work, it may be easier to first output the result to string and measure the line length. In case of line length exceeding some constant, the output for a given object would be regenerated, this time with breaks.

format.conf will almost certainly result in less work in the end. Hardcoded formatting - even relatively smart one - may run into hard to predict edge cases where it will generate giant blocks of indents with syntax or 200+ character lines.
For example json mapgen entries, `iuse_actor`s, inline monster attacks - those are all complex and big. It's not feasible to cover those cases with a simple, predictable formatter without having someone specify the rules explicitly.
Predictability is important because unpredictable formatting results in those extra "format x.json" commits.